htdocs/core/lib/date.lib.php

```diff
@@ -122,6 +122,12 @@ function dol_time_plus_duree($time, $duration_value, $duration_unit, $ruleforend
 {
 	global $conf;
 
+	if ($ruleforendofmonth == 1 && $duration_unit == 'y') {
```

Comment:
duration_unit = y seems supported line 160.
Do you have an example of case that is wrong with current code so we can add it into phpunit test cases ?